java/clients/src/main/java/sleeper/status/report/CompactionJobStatusReport.java

```diff
@@ -0,0 +1,258 @@
+        compactionJobStatusReporter.report(statusList, QueryType.RANGE);
+    }
+
+    private Instant promptForStartRange(Scanner scanner) {
+        Instant startTime = DEFAULT_RANGE_START;
+        while (true) {
+            System.out.printf("Enter start range in format yyyy-MM-dd hh:mm:ss (default is %s):", DEFAULT_RANGE_START.toString());
+            String time = scanner.nextLine();
+            if ("".equals(time)) {
+                System.out.printf("Using default start range %s%n", DEFAULT_RANGE_START);
+                break;
+            }
+            try {
+                Date date = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss").parse(time);
+                startTime = date.toInstant();
+                break;
+            } catch (ParseException e) {
+                System.out.println("Error while parsing input string");
+            }
+        }
+        return startTime;
+    }
+
+    private Instant promptForEndRange(Scanner scanner) {
+        Instant endTime = DEFAULT_RANGE_END;
+
+        while (true) {
+            System.out.printf("Enter end range in format yyyy-MM-dd hh:mm:ss (default is %s):", DEFAULT_RANGE_END.toString());
+            String time = scanner.nextLine();
+            if ("".equals(time)) {
+                System.out.printf("Using default end range %s%n", DEFAULT_RANGE_END);
+                break;
+            }
+            try {
+                Date date = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss").parse(time);
+                endTime = date.toInstant();
+                break;
+            } catch (ParseException e) {
+                System.out.println("Error while parsing input string");
+            }
+        }
+        return endTime;
+    }
+
+    public void handleDetailedQuery(Scanner scanner) {
+        List<String> jobIds;
+
+        System.out.print("Enter jobId to get detailed information about:");
+        String input = scanner.nextLine();
+        if ("".equals(input)) {
+            return;
+        }
+        jobIds = Collections.singletonList(input);
+
+        handleDetailedQuery(jobIds);
+    }
+
+    public void handleDetailedQuery(List<String> jobIds) {
+        List<CompactionJobStatus> statusList = compactionJobStatusCollector.runDetailedQuery(jobIds);
+        compactionJobStatusReporter.report(statusList, QueryType.DETAILED);
+    }
+
+    public void parseQueryParameters(QueryType queryType, String queryParameters) {
+        if (queryType.equals(QueryType.UNFINISHED)) {
+            handleUnfinishedQuery();
+        } else if (queryType.equals(QueryType.DETAILED)) {
+            List<String> jobIds = Collections.singletonList(queryParameters);
+            handleDetailedQuery(jobIds);
+        } else if (queryType.equals(QueryType.RANGE)) {
+            Instant startRange;
+            Instant endRange;
+            try {
+                Date startRangeDate = new SimpleDateFormat("yyyyMMddhhmmss").parse(queryParameters.split(",")[0]);
```

Comment:
`new SimpleDateFormat("yyyyMMddhhmmss")` is probably better as a static variable, especially as it's repeated below?